presto-product-tests/bin/run_on_docker.sh

```diff
@@ -115,41 +115,38 @@ function terminate() {
 }
 
 ENVIRONMENT=$1
+SCRIPT_DIR=$(dirname $(absolutepath "$0"))
+PRODUCT_TESTS_ROOT="${SCRIPT_DIR}/.."
+PROJECT_ROOT="${PRODUCT_TESTS_ROOT}/.."
+ENVIRONMENT_LOCATION="${PRODUCT_TESTS_ROOT}/conf/docker/"
 
-if [[ "$ENVIRONMENT" != "singlenode" && "$ENVIRONMENT" != "multinode" && "$ENVIRONMENT" != "singlenode-kerberized" ]]; then
-   echo "Usage: run_on_docker.sh <singlenode|multinode|singlenode-kerberized> <product test args>"
+if [[ ! -d "$ENVIRONMENT_LOCATION$ENVIRONMENT" ]]; then
+   echo "Usage: run_on_docker.sh <`ls $ENVIRONMENT_LOCATION -I files | tr '\n' '|'`> <product test args>"
    exit 1
 fi
 
 shift 1
 
-SCRIPT_DIR=$(dirname $(absolutepath "$0"))
-PRODUCT_TESTS_ROOT="${SCRIPT_DIR}/.."
-PROJECT_ROOT="${PRODUCT_TESTS_ROOT}/.."
 DOCKER_COMPOSE_LOCATION="${PRODUCT_TESTS_ROOT}/conf/docker/${ENVIRONMENT}/docker-compose.yml"
 DOCKER_PRESTO_VOLUME="/docker/volumes/presto"
+TEMPTO_CONFIGURATION="${DOCKER_PRESTO_VOLUME}/../tempto/tempto-configuration-local.yaml"
 
 PRESTO_SERVICES="presto-master"
 if [[ "$ENVIRONMENT" == "multinode" ]]; then
    PRESTO_SERVICES="${PRESTO_SERVICES} presto-worker"
 fi
 
-TEMPTO_CONFIGURATION="${DOCKER_PRESTO_VOLUME}/presto-product-tests/conf/tempto/tempto-configuration.yaml"
-if [[ "$ENVIRONMENT" == "singlenode-kerberized" ]]; then
-   TEMPTO_CONFIGURATION="${DOCKER_PRESTO_VOLUME}/presto-product-tests/conf/tempto/tempto-configuration-kerberized.yaml"
-fi
-
 # set presto version environment variable
 source "${PRODUCT_TESTS_ROOT}/target/classes/presto.env"
 
 # check docker and docker compose installation
 docker-compose version
 docker version
 
-# stop already running containers
-stop_docker_compose_containers "${PRODUCT_TESTS_ROOT}/conf/docker/singlenode/docker-compose.yml"
-stop_docker_compose_containers "${PRODUCT_TESTS_ROOT}/conf/docker/multinode/docker-compose.yml"
-stop_docker_compose_containers "${PRODUCT_TESTS_ROOT}/conf/docker/singlenode-kerberized/docker-compose.yml"
+for docker_env in `ls $ENVIRONMENT_LOCATION -I files`;
```

Comment:
Maybe extract the `ls $ENVIRONMENT_LOCATION -I files` into `getAvailableEnvironments` functions. You are using that for the second time.